### PR TITLE
Reduce and numeric/bool attributes for the dtype

### PR DESCRIPTION
**Pull request by fjetter**
This adds the two type attributes `_is_numeric` and `_is_boolean` which control some of the pandas internals and are part of the current interface.

The `_reduce` method allows reductions by converting to numpy by default. May be replaced by more efficient functions once available in arrow.

Combined this enables operations like `groupby.sum` (see below). Previously the `B` column would've been omitted

```python
In [1]: import pandas as pd

In [2]: import fletcher as fr

In [3]: df = pd.DataFrame({"A": [1, 1, 2, 2, 3, 3, 1, 4],
   ...:                    "B": fr.FletcherArray(list(range(8))),
   ...:                    "C": [1, 1, 1, 1, 1, 1, 1, 1]})
   ...:

In [4]: df.groupby("A").sum()
Out[4]:
  B  C
A
1 7  3
2 5  2
3 9  2
4 7  1
```

### CODE REVIEW

File: fletcher/base.py [hunk 1]
```diff
@@ -190,6 +202,21 @@ def dtype(self):
         # type: () -> ExtensionDtype
         return self._dtype
 
+    # reduction ops #
+    def _reduce(
+        self, name, axis=0, skipna=True, numeric_only=None, min_count=None, **kwargs
+    ):
+        func = getattr(self, name, None)
+        if func is None:
```
Comment (xhochy) [inline fletcher/base.py]: Could you emit a Warning when we take a numpy function here? NumPy cannot really deal with nulls thus this route is ok but discouraged.

**Comment by artemru** [comment]
are plaining to continue with this PR in near future ? the feature is very natural and is going to be useful in many cases.

**Comment by xhochy** [comment]
@artemru Feel free to have a look at this. The last time I've looked at this, this was introducing new failures to the tests.

**Comment by artemru** [comment]
yes, i'll give it a try. Btw, broken tests were within current fletcher coverage ?

**Comment by xhochy** [comment]
Yes, it broke with the extended tests that came with a new pandas version.

**Comment by xhochy** [comment]
Fixed by #94

**Pull request closed without merging**